compiler/circle2circle/src/Circle2Circle.cpp

```diff
@@ -371,12 +374,39 @@ int entry(int argc, char **argv)
     csv_tokenize(csv_nodes, new_outputs);
   }
 
+  bool dynamic_batch_to_single_batch = false;
+  if (arser["--dynamic_batch_to_single_batch"])
+  {
+    dynamic_batch_to_single_batch = true;
+  }
+
   // Import from input Circle file
   luci::ImporterEx importerex;
   auto module = importerex.importVerifyModule(input_path);
   if (module.get() == nullptr)
     return EXIT_FAILURE;
 
+  // Convert dynamic batch to single batch
+  // Why here? It has to be done before 'optimize', because most optimization
+  // passes are written based on static shapes
+  if (dynamic_batch_to_single_batch)
+  {
+    luci::dynamic_batch_to_single_batch(module.get());
+
+    if (!luci::validate(module.get()))
```

Comment:
We may have to extract shape validation to another method, validate is used for other features too.

Reply:
Looks reasonable. I'll only test shape here.